src/commit.h

```diff
@@ -36,4 +37,20 @@ struct git_commit {
 void git_commit__free(void *commit);
 int git_commit__parse(void *commit, git_odb_object *obj);
 
+typedef enum {
+	GIT_COMMIT_PARSE_QUICK = (1 << 0),
+	GIT_COMMIT_PARSE_ALL   = (1 << 1),
+} git_commit__parse_flags;
+
+int git_commit__parse_commit(git_commit *commit, git_odb_object *obj, git_commit__parse_flags flags);
```

Comment:
I've cut down on the number of flags here because of a bug parsing parents. I think it's better this way, as it makes it less likely you ask for an invalid combination, and we can always split those apart if the need arises.